frontend/src/components/helper/auth-redirect.tsx

```diff
@@ -1,19 +1,19 @@
 import { useNavigate } from "@tanstack/react-router";
-import type { ComponentType } from "react";
 import { useAuthUser } from "~/components/helper/use-auth-user";
 import { Loader } from "~/components/loader";
 
-export function withAuthRedirect(WrappedComponent: ComponentType<any>) {
+import type { ComponentType } from "react";
+
+export function withAuthRedirect(WrappedComponent: ComponentType) {
   return function AuthRedirectWrapper(props: any) {
     const navigate = useNavigate();
 
     const query = useAuthUser();
 
-    if (query.isLoading) {
-      return <Loader />;
-    }
+    if (query.isLoading) return <Loader />;
```

Comment:
This is deliberate choice, inlined conditions are reserved for the JSX, we prefer to put conditions in braces if possible.